src/NuGet.Core/NuGet.Common/PathUtil/PathUtility.cs

```diff
@@ -237,30 +237,39 @@ public static string GetRelativePath(string path1, string path2, char separator)
                 ++index;
             }
 
-            var path = "";
-
             // check if path2 ends with a non-directory separator and if path1 has the same non-directory at the end
             if (len1 + 1 == len2 && !string.IsNullOrEmpty(path1Segments[index]) &&
                 string.Equals(path1Segments[index], path2Segments[index], compare))
             {
-                return path;
+                return string.Empty;
             }
 
+            var path = StringBuilderPool.Shared.Rent(100);
```

Comment:
How this value 100 is chosen here?

Reply:
I changed the value to `256` characters in the latest commit because the input parameters are related to paths whose length may be greater than `100` characters. As I mentioned https://github.com/NuGet/NuGet.Client/pull/4381#discussion_r780523850, If the string builder capacity is greater than 256 characters then the object is not added to the pool, and I assume it will be available for garbage collection immediately. String builder is a suitable candidate here because it reduces string allocations caused by concatenations in two for loops.

Reply:
Did you measure again after recent changes?